File: insert-items.js
const fs = require('fs');
const path = require('path');
const { DynamoDBClient } = require("@aws-sdk/client-dynamodb");
const { DynamoDBDocumentClient, BatchWriteCommand } = require("@aws-sdk/lib-dynamodb");

if (!process.env.TABLE_NAME) {
  console.error("TABLE_NAME environment variable is required");
  process.exit(1);
}

const TABLE_NAME = process.env.TABLE_NAME;
const ITEMS_FILE = "items.json";

async function main() {
  const filePath = path.resolve(ITEMS_FILE);

  if (!fs.existsSync(filePath)) {
    throw new Error(`File not found: ${filePath}`);
  }

  const items = JSON.parse(fs.readFileSync(filePath, 'utf8'));

  const client = new DynamoDBClient({});
  const doc = DynamoDBDocumentClient.from(client);

  const chunkSize = 25;

  for (let i = 0; i < items.length; i += chunkSize) {
    const chunk = items.slice(i, i + chunkSize);

    const requestItems = {
      [TABLE_NAME]: chunk.map(item => ({
        PutRequest: { Item: item }
      }))
    };

    console.log(`Writing batch of ${chunk.length} items...`);

    await doc.send(new BatchWriteCommand({
      RequestItems: requestItems
    }));
  }

  console.log("All country records inserted successfully!");
}

main().catch(err => {
  console.error("ERROR inserting items:", err);
  process.exit(1);
});

File: insert-items.py
import boto3
import json
import math

TABLE_NAME = "CountriesTable"
ITEMS_FILE = "items.json"

def batch_write(table, items):
    with table.batch_writer(overwrite_by_pkeys=["countrycode"]) as batch:
        for item in items:
            batch.put_item(Item=item)

def insert_items():
    dynamodb = boto3.resource("dynamodb")
    table = dynamodb.Table(TABLE_NAME)

    print(f"Reading items from {ITEMS_FILE}...")
    with open(ITEMS_FILE, "r") as f:
        items = json.load(f)

    print(f"Inserting {len(items)} items...")

    batch_size = 25
    total_batches = math.ceil(len(items) / batch_size)

    for i in range(total_batches):
        batch_items = items[i * batch_size : (i + 1) * batch_size]
        print(f"--> Batch {i+1}/{total_batches}: inserting {len(batch_items)} items")
        batch_write(table, batch_items)

    print("All items inserted successfully!")

if __name__ == "__main__":
    insert_items()
